packages/pubsub/__tests__/AWSAppSyncRealTimeProvider.test.ts

```diff
@@ -0,0 +1,879 @@
+					});
+
+					const subscription = observer.subscribe({});
+
+					await fakeWebSocketInterface?.readyForUse;
+					await fakeWebSocketInterface?.triggerOpen();
+					await fakeWebSocketInterface?.handShakeMessage();
+					await fakeWebSocketInterface?.sendDataMessage({
+						type: MESSAGE_TYPES.GQL_DATA,
+						payload: { data: {} },
+					});
+					await subscription.unsubscribe();
+					await fakeWebSocketInterface?.triggerClose();
+
+					expect(fakeWebSocketInterface.hasClosed).resolves.toBe(undefined);
+				});
+
+				test('failure to ack before timeout', async () => {
+					expect.assertions(1);
+
+					await replaceConstant('START_ACK_TIMEOUT', 20, async () => {
+						const observer = provider.subscribe('test', {
+							appSyncGraphqlEndpoint: 'ws://localhost:8080',
+						});
+
+						const subscription = observer.subscribe({
+							error: () => {},
+						});
+
+						await fakeWebSocketInterface?.readyForUse;
+						await fakeWebSocketInterface?.triggerOpen();
+						await fakeWebSocketInterface?.handShakeMessage();
+
+						// Wait long enough that the shortened timeout will elapse
+						await fakeWebSocketInterface?.wait(100);
+
+						expect(loggerSpy).toBeCalledWith(
+							'DEBUG',
+							'timeoutStartSubscription',
+							expect.anything()
+						);
+					});
+				});
+
+				test('connection init timeout', async () => {
+					expect.assertions(1);
+
+					await replaceConstant('CONNECTION_INIT_TIMEOUT', 20, async () => {
+						const observer = provider.subscribe('test', {
+							appSyncGraphqlEndpoint: 'ws://localhost:8080',
+						});
+
+						const subscription = observer.subscribe({
+							error: () => {},
+						});
+
+						await fakeWebSocketInterface?.readyForUse;
+						await fakeWebSocketInterface?.triggerOpen();
+
+						// Wait long enough that the shortened timeout will elapse
+						await fakeWebSocketInterface?.wait(100);
+
+						// Watching for raised exception to be caught and logged
+						expect(loggerSpy).toBeCalledWith(
+							'DEBUG',
+							'error on bound ',
+							expect.objectContaining({
+								message: expect.stringMatching(
+									'Connection timeout: ack from AWSRealTime'
+								),
+							})
+						);
+					});
+				});
+
+				describe('constructed against the different auth interfaces', () => {
+					test('authenticating with API_KEY', async () => {
+						expect.assertions(1);
+
+						const subscription = provider
+							.subscribe('test', {
+								appSyncGraphqlEndpoint: 'ws://localhost:8080',
+								authenticationType: 'API_KEY',
+							})
+							.subscribe({});
+						await fakeWebSocketInterface?.readyForUse;
+
+						expect(loggerSpy).toBeCalledWith(
+							'DEBUG',
+							'Authenticating with API_KEY'
+						);
+					});
+
+					test('authenticating with AWS_IAM', async () => {
+						expect.assertions(1);
+
+						jest.spyOn(Credentials, 'get').mockImplementation(() => {
+							return Promise.resolve({});
+						});
```

Comment:
Nit: You should be able to use `mockResolvedValue` or `mockRejectedValue` to make this (and other places this pattern appears) more readable 🍬 
```suggestion
						jest.spyOn(Credentials, 'get').mockResolvedValue({});
```